### ISSUE TITLE
Segfault on SoundRecorder.start()

### ISSUE THREAD

**Issue by zyedidia** [SFML/SFML#902]
I have copied this code from the documentation [here] (http://www.sfml-dev.org/documentation/2.3/classsf_1_1SoundRecorder.php) and I am getting a segmentation fault when `recorder.start()` is called:

```Cpp
#include <SFML/Audio.hpp>

class CustomRecorder : public sf::SoundRecorder
{
	virtual bool onStart() // optional
	{
		// Initialize whatever has to be done before the capture starts
		// Return true to start playing
		return true;
	}
	virtual bool onProcessSamples(const sf::Int16* samples, std::size_t sampleCount)
	{
		// Do something with the new chunk of samples (store them, send them, ...)
		// Return true to continue playing
		return true;
	}
	virtual void onStop() // optional
	{
		// Clean up whatever has to be done after the capture ends
	}
};

int main() {
	// Usage
	if (CustomRecorder::isAvailable())
	{
		CustomRecorder recorder;
		if (!recorder.start()) // Segfault happens here
			return -1;

		recorder.stop();
	}
	return 0;
}
```

**Comment by binary1248** [comment]
Duplicate of #900.

**Comment by mantognini** [comment]
@zyedidia On which system did you experience this? Mac?

I know this is old, but I've a doubt whether it's really a duplicate or not.

**Comment by zyedidia** [comment]
Yeah I experienced this issue on Mac. At the time I was on Mavericks, but now I am on El Capitan. When I run the code now I get the following message:

```
WARNING:  140: This application, or a library it uses, is using the deprecated Carbon Component Manager for hosting Audio Units. Support for this will be removed in a future release. Also, this makes the host incompatible with version 3 audio units. Please transition to the API's in AudioComponent.h.
```

Though it does seem to work despite the warning.

**Comment by mantognini** [comment]
Ok, thanks for the quick reply! This is really a duplicate then. Also, it's still crashed on 10.10.

See #1000 and its fix #1057 for the warnings.
